### PR TITLE
Loosen description of `last_modified`

### LINKED ISSUE #41: Semantics of `last_modified` unclear
In the case where `last_modified` is called with a key that points to a dictionary it returns the latest `last_modified` value of its direct children.

https://github.com/mirage/mirage-kv/blob/1e3cdbd8167193de8a78b924fd3ccdf3f09c4afa/src/mirage_kv.mli#L159-L165

This definition is not well-founded as a dictionary's direct children could all be dictionaries themselves.

The interface doesn't explicitly disallow empty dictionaries although they can not be created directly. Depending on the implementation it could be achieved by setting a key `/empty/to-be-removed` and them removing `/empty/to-be-removed`. The interface doesn't mandate removing empty dictionaries resulting from `remove`.

A way out could be to default to `Ptime.min`, or leave it up to the implementation as some implementations have a concept of timestamps for dictionaries/directories.

### PR DESCRIPTION
**Pull request by reynir**
Fixes #41.

### CODE REVIEW

**Review by hannesm — APPROVED** [review]
looks good

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+### Unreleased
+
+* Leave it up to implementations how to interpret `last_modified` for
+  dictionaries, or even not implement it (then returning ``Error
+  (`Value_expected _)``. The previous definition was not well founded when the
+  dictionary doesn't contain any values directly, and some implementations
+  implements `last_modified` differently from the description on dictionaries.
```
Comment (hannesm) [inline CHANGES.md:7]:
```suggestion
  implements `last_modified` differently from the description on dictionaries.
  (reported in #41, fixed in #42 by @reynir)
```

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,11 @@
+### Unreleased
+
+* Leave it up to implementations how to interpret `last_modified` for
+  dictionaries, or even not implement it (then returning ``Error
+  (`Value_expected _)``. The previous definition was not well founded when the
+  dictionary doesn't contain any values directly, and some implementations
```
Comment (hannesm) [inline CHANGES.md:6]:
```suggestion
  dictionary doesn't contain any values directly (or values were removed), and some implementations
```

File: src/mirage_kv.mli [hunk 3]
```diff
@@ -160,9 +160,9 @@ module type RO = sig
   (** [last_modified t k] is the last time the value bound to [k] in
      [t] has been modified.
 
-      When the value bound to [k] is a dictionary, the modification
-     time is the latest modification of all entries in that
-     dictionary. This behaviour is only one level deep and not recursive. *)
+      When the value bound to [k] is a dictionary, the implementation is free
+      to decide how to compute a last modified timestamp, or return [Error
+      (`Value_expected _)]. *)
```
Comment (hannesm) [inline src/mirage_kv.mli:165]:
```suggestion
      (`Value_expected _)]. Earlier, the last-modified of a dictionary was defined as the latest modification
      in the dictionary (up to a depth of 1), but that didn't take value removal into account, and is neither what a
      Unix system provides. *)
```

File: CHANGES.md (hunk 2 above)
Comment (reynir) [reply CHANGES.md:6]: I think it is okay to not describe how this condition can occur.

Comment (hannesm) [reply CHANGES.md:6]: fine with me

File: src/mirage_kv.mli (hunk 3 above)
Comment (reynir) [reply src/mirage_kv.mli:165]: I am a bit hesitant about this. I feel it belongs in the CHANGES file. How long should we keep a description of the old behavior in the docs? WDYT?

Comment (hannesm) [reply src/mirage_kv.mli:165]: I agree, let's put the change in the changes, and keep the docstring brief to elaborate on the current interface.

**Pull request merged**